internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -596,20 +596,20 @@ func (b *BeatsMonitor) injectMetricsInput(cfg map[string]interface{}, componentI
 				idKey: "metrics-monitoring-" + name,
 				"data_stream": map[string]interface{}{
 					"type":      "metrics",
-					"dataset":   fmt.Sprintf("elastic_agent.%s", name),
+					"dataset":   fmt.Sprintf("elastic_agent.%s", binaryName),
```

Comment:
We did do this change before as well: https://github.com/elastic/elastic-agent/pull/1854/files

Could you just change `name := strings.ReplaceAll(strings.ReplaceAll(unit, "-", "_"), "/", "_") // conform with index naming policy` above instead to fix this?

Specifically using `strings.ReplaceAll(binaryName` instead of `strings.ReplaceAll(unit`

Reply:
Here is this suggested change as a forward-port PR: https://github.com/elastic/elastic-agent/pull/2006

Reply:
Hmm I find doing it this way curious. I believe it will result in multiple streams with the same `id`, which I presumed would be an issue.

Reply:
> I believe it will result in multiple streams with the same idKey, which I presumed would be an issue.

I'm not very familiar with the constraints on index names here, but the binary name can be anything that is a valid file name so I could believe we have to do some sanitization/normalization on it.

We did the exact same change for logs on main and 8.6 in https://github.com/elastic/elastic-agent/pull/1845/files so if we decide to change it we should change it in both places.

Reply:
Merged the backport to fix main quickly. We can still adjust the way this is done if needed.